Node.js Port

ulissesferreira opened the pull request:
- [x] Display other online users
- [x] Add log-out option
- [x] Change password method to environment variable instead of file content
- [ ] Harden security (POST method using a hash instead of the plaintext password to avoid sniffing attacks)
- [ ] Harden security 2 (We are vulnerable to bruteforcing - add some way to stop infinite submits)
- [ ] Harden security 3 (We can crash the server if we send enough data through the text fields. Limit the size)
- [ ] Improve design (after the MVP checks out)

ulissesferreira committed:
NodeJS port development started! First commit

ulissesferreira committed:
First source commit. Basic login port

ulissesferreira committed:
First source commit.

hacdias commented on README.md:
```diff
@@ -1,12 +1,24 @@
 ![Who's there?](https://cloud.githubusercontent.com/assets/5447088/22626990/14522f80-ebb1-11e6-9b40-9a027ed3478d.gif)
 
+<<<<<<< HEAD
```
You forgot to solve this merge conflict.

hacdias commented on css/styles.css:
```diff
@@ -0,0 +1,19 @@
+body {
+  background: skyblue; font-family: verdana; color: #fff; padding: 30px
```
Please break this into multiple lines to be more legible. 😮

ulissesferreira committed:
fix: last commit was weird

hacdias reviewed (changes requested):
~One more thing (despite the comments): delete the `node_modules` folder and add the following lines to .gitignore:~

```
node_modules
*.lock
```

ulissesferreira committed:
Removed the folder

ulissesferreira committed:
fix: stupid commit errors

hacdias committed:
remove package lock

add linting

travis

update pkg json and add start script

Fix tests not running directly

ulissesferreira committed:
Vejam o erro

hacdias committed:
store users on file

hacdias committed:
fix files

hacdias committed:
breaking changes

ulissesferreira committed:
Changed to ws - server side

hacdias committed:
auto solve some issues

hacdias committed:
some small fixes

ulissesferreira committed:
Now using WebSockets. Logout is disabling the socket though, need fix Javascript Standard Style

ulissesferreira committed:
Now using WebSockets. Logout is disabling the socket though, need fix Javascript Standard Style

ulissesferreira committed:
All fixed. Missing: JavaScript Standard

hacdias committed:
feat: pretty

hacdias commented:
@UlissesFerreira,

- [ ] Improve styles (mainly the list of users)
- [ ] Add comments to the code (in English) (don't add too many)
- [ ] Put the websocket in a different path in the same port of the server.

hacdias committed:
feat: use websocket on the same port

hacdias committed:
add blank line

hacdias commented on README.md:
````diff
@@ -1,12 +1,15 @@
 There must be a ```thisisnotthepassword.txt``` file in the same place where the binary is being executed and it must contain the password that'll be used by the members to change their status.
 
-You can change the port and activate/deactivate the SSL with the flags ```--port``` and ```--secure```, respectively.
+## NPM Packages
````
Remove this section. If anyone wants to know the packages we use, they can just look at the `package.json`.

hacdias commented on README.md:
```diff
@@ -1,12 +1,15 @@
 ![Who's there?](https://cloud.githubusercontent.com/assets/5447088/22626990/14522f80-ebb1-11e6-9b40-9a027ed3478d.gif)
 
-[![Build](https://img.shields.io/travis/upframe/whosthere.svg?style=flat-square)](https://travis-ci.org/upframe/whosthere)
```
The Travis checking shouldn't be removed.

hacdias commented on README.md:
````diff
@@ -1,12 +1,15 @@
 
 ## Usage
 
 There must be a ```thisisnotthepassword.txt``` file in the same place where the binary is being executed and it must contain the password that'll be used by the members to change their status.
````
Change this and explain the environment variables they have to set and the defaults.

malikpiara committed:
Updated src css with Upframe color scheme.

hacdias commented:
What's the state of this @ulissesferreira?

hacdias commented on src/public/css/styles.css:
```diff
@@ -35,26 +34,30 @@ div.user {
+  border-radius: 3px;
   outline: none;
   cursor: pointer;
+  font-family: roboto;
```
You don't import Roboto so this won't work unless you've got the font installed on your computer.

ulissesferreira committed:
fix: Roboto not imported

ulissesferreira committed:
docs: install and usage explanation

ulissesferreira committed:
docs: fixed typos, sorry guys

ulissesferreira committed:
docs: fixed typos... again...

ulissesferreira committed:
docs: Add Travis integration

ulissesferreira committed:
docs: readme javascript quality badge + add some comments throughout the code

ulissesferreira committed:
docs: fix readme javascript quality badge

ulissesferreira committed:
docs: readme badges with same style

hacdias committed:
Update README.md

hacdias committed:
Update README.md

hacdias committed:
style: apply Roboto to everything

hacdias committed:
updates
